### PR TITLE
[Estuary] display metadata in addon info dialog + show origin icons in addonbrowser

### PR DESCRIPTION
**Pull request by ronie**
## Description

This PR adds some additional addon meta-data to the addoninfo dialog in Estuary.
(this also makes the addoninfo dialog layout more consistent with the music/video infodialogs)

please do not merge before https://github.com/xbmc/xbmc/pull/18381 gets in.

@phunkyfish / @howie-f 

## Screenshots (if appropriate):
before:
b1
b2

after:
a1
a2

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by howie-f — APPROVED** [review]
looks good, thanks

**Review by phunkyfish — APPROVED** [review]
👍

**Comment by ronie** [comment]
@phunkyfish @howie-f the second commit in this PR adds the addon origin icons to the addonbrowser.

example of the 'official' / 'third party repo' / 'manual' icons: 

origin

**Comment by phunkyfish** [comment]
Nice, thanks @ronie, I like it.

Would it be possible to also add the icon next to the origin text on the addon info dialog as each origin type has an icon now?

**Review by howie-f — APPROVED** [review]
yes, looks nice :+1:

**Comment by ronie** [comment]
added 'pinned' variations of the icons as well

pinned

**Comment by phunkyfish** [comment]
> added 'pinned' variations of the icons as well
> 
> pinned

Nice 😃

**Comment by phunkyfish** [comment]
`display` is spelled incorrectly in first commit message

**Comment by howie-f** [comment]
yes, cool

**Comment by phunkyfish** [comment]
Any chance we can also have those respective icons next to the origin description text on the AddonInfo dialog too?

**Comment by ronie** [comment]
> That would be the icing ;)

i'd rather reserve that space for the repository name. some repo names are obviously longer than ours.
also i think it doesn't add much to display the same info both in text and as an icon.

**Comment by phunkyfish** [comment]
> > That would be the icing ;)
> 
> i'd rather reserve that space for the repository name. some repo names are obviously longer than ours.
> also i think it doesn't add much to display the same info both in text and as an icon.

So how do you know from the addon info dialog it an addon is from an official or private repo?

**Comment by jjd-uk** [comment]
> So how do we let the user know from the addon info dialog if an addon is from an official or private repo?

You could use the infolabel **ListItem.Property(Addon.IsFromOfficialRepo)** then have a new line under **Origin** for something like:

**Official repo: Yes/No**

**Comment by phunkyfish** [comment]
Because of the new pinning logic I think we need to display both whether or not the version is pinned/locked and if it's from an official repo or not on the addon info dialog. I'm not sure the best way to do this but they are both useful piece of info for the user.

**Comment by garbear** [comment]
Should the pngs be minified? I see that the replacement png in the diff has the properties:

Old:

W: 32px | H: 32px | Size: 378 Bytes

New:

W: 32px | H: 32px | Size: 1.79 KB

**Comment by ronie** [comment]
> Should the pngs be minified?

thanx for the reminder!
i always forget Gimp adds a full blown set of metadata to images by default.

**Comment by ronie** [comment]
> So how do we let the user know from the addon info dialog if an addon is from an official or private repo?

they can tell from the repository name?

**Comment by phunkyfish** [comment]
Rebrands could have a number of repos so not so much. Also you could name your repo similar to kodi’s and have it look official.

**Comment by phunkyfish** [comment]
@ronie how about the space under the main icon on the left? If that is not in use it could be used for official/pinned.

**Comment by ronie** [comment]
after trying several options and not liking any of them, i settled for this:

done

**Comment by phunkyfish** [comment]
I tried to build (I rebased against master) but everything I see is pinned in the browser and I don't see the any icon in the info dialog :(

**Comment by howie-f** [comment]
i cherry-picked the commit and tested with estuary. i can see the pinned icon in the info-dialog (with a little delay)
in addon-directory i have to scroll the estuary out of screen and back in for the pinned to appear

**Comment by phunkyfish** [comment]
Let me try again.

**Comment by ronie** [comment]
i've squashed the two commits and rebased on master (no idea if it was needed).
it all works as expected on my end.

**Comment by howie-f** [comment]
hm, i'm still getting refresh issues... what i am doing in the followong screenshots is just moving the "selected" bar one item down.
please notice the lock-symbol @ estuary

can be that deb stretch is just too old.. 

screenshot000

screenshot001

**Comment by phunkyfish** [comment]
Now that I have correctly cherry-picked the commit it works perfectly for me. I do not see your issue @howie-f

**Comment by phunkyfish** [comment]
@howie-f maybe try to build @ronie ’a branch directly now that it’s been rebased.

Bear in mind that @AlwinEsch has updated some addons stuff so you may need to rebuild depends and any addons.

**Comment by ronie** [comment]
> i cherry-picked the commit and tested with estuary. i can see the pinned icon in the info-dialog (with a little delay)
> in addon-directory i have to scroll the estuary out of screen and back in for the pinned to appear.

@howie-f i can't reproduce and it's not something i can explain from the skin point of view either.
the pinned/unpinned state of the icon depends on the ListItem.IsAutoUpdateable infobool.

maybe someone can double-check if the infobool was implemented correctly: https://github.com/xbmc/xbmc/pull/18381/commits/c16ccf40c75b198e1ab19248d1f200e0639b63e5

**Comment by howie-f** [comment]
@ronie thanks, as you both confirmed positive this shouldnt prevent the pr. i have a slight idea why it‘s happening but will not come to check it before evening today

**Comment by ronie** [comment]
i'm not in a hurry to get this merged and rather be save than sorry.
so take your time and once you are 100% sure it's not something on my end, lmk.

**Comment by phunkyfish** [comment]
I can reproduce @howie-f's issue. @howie-f you might post the video of it here.

With auto-updates enabled:

1) Go to `Look and feel` -> `Skins` -> `Estouchy`
2) In the addon info dialog disable auto-updates and the icon next origin will gain a pin
3) Exit back to the addon browser and the icon there will be missing the pin
4) Exit back to `Look and feel` and then back to `Skins`, voila, the pin is back.

So there appears to be a refresh missing when you return to the addon browser from the addon info dialog.

**Comment by phunkyfish** [comment]
@ronie do you know how to get the add-on browser to refresh when the add-on info dialog is closed?

Or happen to now who would know how to do this?

**Comment by ronie** [comment]
not really, but if you use the enable/disable button in the addoninfo dialog, the listing in the addon browser gets updated correctly.
perhaps someone can follow that code path... and see what's missing with regards to the auto-update button?

**Comment by howie-f** [comment]
https://github.com/xbmc/xbmc/blob/cdfe287bb58dfb75de8cbce591824d449dcee0c5/xbmc/addons/AddonManager.cpp#L739
@phunkyfish i think your `m_events.Publish()` hint is correct. i will look into it one my build env is up again (c++17 pr knocked it out 😏)

**Comment by phunkyfish** [comment]
> not really, but if you use the enable/disable button in the addoninfo dialog, the listing in the addon browser gets updated correctly.
> perhaps someone can follow that code path... and see what's missing with regards to the auto-update button?

Cool, nice hint, thanks!

**Comment by phunkyfish** [comment]
@howie-f I think if you publish this event: `AddonEvents::MetadataChanged` when when either auto-update is toggled or there is an update it should redraw the UI. It's only used by the directory provider which is what we want. If it works we can then create a new event type (cause we're not actually changing the metadata).

**Comment by phunkyfish** [comment]
@ronie you can merge this PR if you'd like. We can apply the UI refresh once it's in.

**Comment by ronie** [comment]
not sure if this is correct, but it fixes the issue:
https://github.com/ronie/xbmc/commit/ac1bca6aa72a6aa673bec66c356ac27454c97c41

**Comment by phunkyfish** [comment]
Cool. So any event must update the UI.

I would just have a single event: `UpdateRuleModified`. Separate enabled/disabled events seem overkill.

**Comment by howie-f** [comment]
cool 👍🏻 @ronie would you add it please?

**Comment by howie-f** [comment]
cool 👍🏻 @ronie would you add it please?

**Comment by howie-f** [comment]
cool 👍🏻 @ronie would you add it please?

**Comment by howie-f** [comment]
cool 👍🏻 @ronie would you add it please?

**Comment by ronie** [comment]
done: https://github.com/xbmc/xbmc/pull/18447
